src/js/media-fullscreen-button.ts

```diff
@@ -140,6 +151,18 @@ class MediaFullscreenButton extends MediaChromeButton {
       new globalThis.CustomEvent(eventName, { composed: true, bubbles: true })
     );
   }
+
+  private moveFocusToMedia(): void {
+    const mediaController = getMediaController(this);
+    if (!mediaController) return;
+
+    const mediaElement = mediaController.querySelector(
+      '[slot="media"]'
+    ) as HTMLElement | null;
+    if (!mediaElement) return;
+
+    mediaElement.focus();
+  }
```

Comment:
This code shouldn't live in the button component but in the state mediator. You can pass the event type as a detail to the MediaUIEvents.MEDIA_ENTER_FULLSCREEN_REQUEST event and then if it's pointer do the focus in the mediaIsFullscreen.set function in state-mediator.ts